code/modules/random_map/noise/noise.dm

```diff
@@ -167,4 +168,33 @@
 				else if(map[current_cell] < total)
 					map[current_cell]-=cell_smooth_amt
 				map[current_cell] = max(0,min(cell_range,map[current_cell]))
-		map = next_map
\ No newline at end of file
+		map = next_map
+	
+	if(smooth_single_tiles)
+		var/list/buddies = list()
+		for(var/x in 1 to limit_x - 1)
+			for(var/y in 1 to limit_y - 1)
+				var/mapcell = get_map_cell(x,y)
+				var/list/neighbors = get_neighbors(x, y)
+				buddies.Cut()
+				for(var/cell in neighbors)
+					if(noise2value(map[cell]) == noise2value(map[mapcell]))
+						buddies |= cell
+				if(!length(buddies))
+					map[mapcell] = map[pick(neighbors)]
+
+/datum/random_map/noise/proc/get_neighbors(x, y, include_diagonals)
+	. = list()
+	if(!include_diagonals)
+		for(var/list/offset in list(list(-1, 0), list(1, 0), list(0, 1), list(0,-1)))
```

Comment:
How hot is this proc? Might be good to break this into a static local var just so you don't need to init 5 lists every time.

Reply:
Done